source/backend/dal/Repositories/DispositionFileRepository.cs

```diff
@@ -325,6 +325,26 @@ public PimsDispositionSale GetDispositionFileSale(long dispositionId)
                 .Where(x => x.DispositionFileId == dispositionId).FirstOrDefault();
         }
 
+        public PimsDispositionSale AddDispositionFileSale(PimsDispositionSale dispositionSale)
+        {
+            Context.PimsDispositionSales.Add(dispositionSale);
+
+            return dispositionSale;
+        }
+
+        public PimsDispositionSale UpdateDispositionFileSale(long saleId, PimsDispositionSale dispositionSale)
```

Comment:
q: does the dispositionSaleId not valid here? why is the saleId prefered?

Reply:
In general we want to think of these methods as being independent from their calling services. The way this method is written, there is a hidden requirement that the saleId should match the dispositionSale.Id which is not guaranteed. The hidden requirement is met by the one service calling this method, but future services may want to use it who may not be doing this validation.

Also, In general, reducing a methods parameters to only what is required is a good rule of thumb.